common/src/main/java/com/wynntils/models/items/encoding/impl/block/PowderDataTransformer.java

```diff
@@ -137,11 +137,23 @@ private ErrorOr<PowderData> decodePowderData(ArrayReader<UnsignedByte> byteReade
             }
 
             // Decode the powder value
-            int element = powderValue / 6;
-            int tier = powderValue % 6;
+            // if the powder value is 0, it means no powder is present, thus no data is
+            // decoded for it
+            if (powderValue != 0) {
+                int element = powderValue / 6;
+                int tier = powderValue % 6;
+
+                // if the powder value is divisible by 6, it means that we have a tier 6 powder
+                if (powderValue % 6 == 0) {
+                    // subtract 1 from the element to get the correct element
+                    element--;
```

Comment:
Is this actually correct? As in, previously, element has been subtracted by 1 in every case, and it worked.

Reply:
powders are encoded as `tier * 6` which means that 1 must only be subtracted if the powder is of tier 6. If 6 were to be subtracted from say a powder of tier 5 it would instead change the type to be 1 lower or put the value into the negatives.

The code previously works as wynntils only ever encodes items as having tier 6 powders. If an encode with a tier lesser than 6 is generated and put thru the previous code it does what I described above.